src/html/fetch-content.js

```diff
@@ -27,34 +27,21 @@ async function fetchContent(context, {
   const branchOrRef = branch || ref || 'master';
   const refOrBranch = ref || branch || 'master';
 
-  // compute content proxy url
-  let contentProxyUrl = secrets.CONTENT_PROXY_URL || '';
-  if (!contentProxyUrl) {
-    if (resolver) {
-      contentProxyUrl = resolver.createURL({
-        package: 'helix-services',
-        name: 'content-proxy',
-        version: 'v2',
-      });
-    } else {
-      contentProxyUrl = versionLock.createActionURL({
-        name: 'content-proxy@v2',
-        packageName: 'helix-services',
-        namespace: process.env.__OW_NAMESPACE || 'helix',
-      });
-    }
+  let contentProxyUrl;
+  if (secrets.CONTENT_PROXY_URL) {
+    contentProxyUrl = new URL(secrets.CONTENT_PROXY_URL);
+    contentProxyUrl.searchParams.append('owner', owner);
+    contentProxyUrl.searchParams.append('repo', repo);
+    contentProxyUrl.searchParams.append('path', path);
+    contentProxyUrl.searchParams.append('ref', refOrBranch); // prefer ref for content fetching
+  } else {
+    contentProxyUrl = new URL(`https://${branchOrRef}--${repo}--${owner}.hlx.page${path}`);
```

Comment:
branch is preferred here...I guess. I think since most of the content is not in github anymore, this doesn't make a big difference.

Reply:
No, I would use `ref` over branch in any case. It makes things faster and easier to purge.